addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -122,6 +122,41 @@ def _get_report_data(self, bom_id, searchQty=0, searchVariant=False):
             'precision': self.env['decimal.precision'].precision_get('Product Unit of Measure'),
         }
 
+    @api.model
+    def _get_components_closest_forecasted(self, lines, line_quantities, parent_bom, product_info, ignore_stock=False):
+        """
+            Returns a dict mapping a product to the closest date in the forecast report where
+            consummed quantity >= forecasted quantity.
+        """
+        if ignore_stock:
+            return {}
+        closest_forecasted = {}
+        remaining_products = self.env['product.product']
+        product_quantities_info = {}
+        for line in lines:
+            product = line.product_id
+            quantities_info = product_quantities_info.setdefault(product.id, self._get_quantities_info(product, line.product_uom_id, parent_bom, product_info))
+            stock_loc = quantities_info['stock_loc']
+            product_info[product.id]['consumptions'][stock_loc] += line_quantities.get(line.id, 0.0)
+            if product.detailed_type != 'product' or quantities_info['stock_loc'] != 'in_stock':
+                # Use date.min as a sentinel value for _get_stock_availability
+                closest_forecasted[product.id] = date.min
```

Comment:
```suggestion
            if product.type != 'product':  # [1]
                # Use date.min as a sentinel value for _get_stock_availability
                closest_forecasted[product.id] = date.min
            elif quantities_info['stock_loc'] != 'in_stock':  # [2]
                closest_forecasted[product.id] = date.max
```
[1] Ideally we'd use `type`, but I used the wrong one all around this report, so maybe for consistency's sake... :thinking: 

[2] If in the same condition, then all subcontracted locations will always display components as *Available*, instead of checking whether there is enough stock in that location. Setting it to the equivalent of *Unavailable* instead makes it follow the current flow, which will display the resupply delay.

Reply:
[1] I used `detailed_type` here to be consistent with the condition in `_get_stock_availability` in the current version. I could change it to `type` but I'd have to change it everywhere in `mrp_report_bom_structure.py` and I'm not sure that it should be done in this PR 🤔

[2] Yup, I misread the condition in `_get_stock_availability`

Reply:
> [1] [...] I'm not sure that it should be done in this PR :thinking:

That's what I told myself when I wrote the first point. :slightly_smiling_face: 
We can keep it like this for now as for `product`, `type` and `detailed_type` are the same.